src/helpers/helpers.easing.js

```diff
@@ -1,194 +1,84 @@
 
-  easeInOutQuint(t) {
-    if ((t /= 0.5) < 1) {
-      return 0.5 * t * t * t * t * t;
-    }
-    return 0.5 * ((t -= 2) * t * t * t * t + 2);
-  },
+  easeInOutQuint: t => ((t /= 0.5) < 1)
+    ? 0.5 * t * t * t * t * t
+    : 0.5 * ((t -= 2) * t * t * t * t + 2),
 
-  easeInSine(t) {
-    return -Math.cos(t * HALF_PI) + 1;
-  },
+  easeInSine: t => -Math.cos(t * HALF_PI) + 1,
 
-  easeOutSine(t) {
-    return Math.sin(t * HALF_PI);
-  },
+  easeOutSine: t => Math.sin(t * HALF_PI),
 
-  easeInOutSine(t) {
-    return -0.5 * (Math.cos(PI * t) - 1);
-  },
+  easeInOutSine: t => -0.5 * (Math.cos(PI * t) - 1),
 
-  easeInExpo(t) {
-    return (t === 0) ? 0 : Math.pow(2, 10 * (t - 1));
-  },
+  easeInExpo: t => (t === 0) ? 0 : Math.pow(2, 10 * (t - 1)),
 
-  easeOutExpo(t) {
-    return (t === 1) ? 1 : -Math.pow(2, -10 * t) + 1;
-  },
+  easeOutExpo: t => (t === 1) ? 1 : -Math.pow(2, -10 * t) + 1,
 
-  easeInOutExpo(t) {
-    if (t === 0) {
-      return 0;
-    }
-    if (t === 1) {
-      return 1;
-    }
-    if ((t /= 0.5) < 1) {
-      return 0.5 * Math.pow(2, 10 * (t - 1));
-    }
-    return 0.5 * (-Math.pow(2, -10 * --t) + 2);
-  },
+  easeInOutExpo: t => atEdge(t) ? t : t < 0.5
+    ? 0.5 * Math.pow(2, 10 * (t * 2 - 1))
+    : 0.5 * (-Math.pow(2, -10 * (t * 2 - 1)) + 2),
 
-  easeInCirc(t) {
-    if (t >= 1) {
-      return t;
-    }
-    return -(Math.sqrt(1 - t * t) - 1);
-  },
+  easeInCirc: t => (t >= 1) ? t : -(Math.sqrt(1 - t * t) - 1),
 
-  easeOutCirc(t) {
-    return Math.sqrt(1 - (t -= 1) * t);
-  },
+  easeOutCirc: t => Math.sqrt(1 - (t -= 1) * t),
 
-  easeInOutCirc(t) {
-    if ((t /= 0.5) < 1) {
-      return -0.5 * (Math.sqrt(1 - t * t) - 1);
-    }
-    return 0.5 * (Math.sqrt(1 - (t -= 2) * t) + 1);
-  },
+  easeInOutCirc: t => ((t /= 0.5) < 1)
+    ? -0.5 * (Math.sqrt(1 - t * t) - 1)
+    : 0.5 * (Math.sqrt(1 - (t -= 2) * t) + 1),
 
-  easeInElastic(t) {
-    let s = 1.70158;
-    let p = 0;
-    let a = 1;
-    if (t === 0) {
-      return 0;
-    }
-    if (t === 1) {
-      return 1;
-    }
-    if (!p) {
-      p = 0.3;
-    }
-    if (a < 1) {
-      a = 1;
-      s = p / 4;
-    } else {
-      s = p / TAU * Math.asin(1 / a);
-    }
-    return -(a * Math.pow(2, 10 * (t -= 1)) * Math.sin((t - s) * TAU / p));
-  },
+  easeInElastic: t => atEdge(t) ? t : elasticIn(t, 0.075, 0.3),
 
-  easeOutElastic(t) {
-    let s = 1.70158;
-    let p = 0;
-    let a = 1;
-    if (t === 0) {
-      return 0;
-    }
-    if (t === 1) {
-      return 1;
-    }
-    if (!p) {
-      p = 0.3;
-    }
-    if (a < 1) {
-      a = 1;
-      s = p / 4;
-    } else {
-      s = p / TAU * Math.asin(1 / a);
-    }
-    return a * Math.pow(2, -10 * t) * Math.sin((t - s) * TAU / p) + 1;
-  },
+  easeOutElastic: t => atEdge(t) ? t : elasticOut(t, 0.075, 0.3),
 
   easeInOutElastic(t) {
-    let s = 1.70158;
-    let p = 0;
-    let a = 1;
-    if (t === 0) {
-      return 0;
-    }
-    if ((t /= 0.5) === 2) {
```

Comment:
interesting that this old code didn't use `if (t === 1)`

Reply:
yeah, the functions were/are quite interestingly made. a lot of assignments used inside calculations.